format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java

```diff
@@ -117,32 +118,40 @@ public Schema getDefaultSchema(@Nullable FormatContext context) throws IOExcepti
     String filePath = conf.getProperties().getProperties().getOrDefault("path", null);
     SeekableInput seekableInput = null;
     FileReader<GenericRecord> dataFileReader = null;
+    GenericRecord firstRecord = null;
```

Comment:
better to declare things closer to where they are used. In this case, SeekableInput, FileReader, and GenericRecord  can all be declared within the loop instead of outside of it.